runtime/doc/api.txt

```diff
@@ -2400,25 +2402,27 @@ nvim_buf_set_text({buffer}, {start_row}, {start_col}, {end_row}, {end_col},
                   {replacement})
                 Sets (replaces) a range in the buffer
 
-                This is recommended over nvim_buf_set_lines when only
+                This is recommended over |nvim_buf_set_lines()| when only
                 modifying parts of a line, as extmarks will be preserved on
                 non-modified parts of the touched lines.
 
-                Indexing is zero-based and end-exclusive.
+                Row indices are zero-based and end-inclusive. Column indices
+                are zero-based, end-exclusive, and correspond to bytes, not
+                character columns (see |col()| vs. |charcol()|).
 
-                To insert text at a given index, set `start` and `end` ranges
-                to the same index. To delete a range, set `replacement` to an
-                array containing an empty string, or simply an empty array.
+                To insert text at a given `(row, column)` location, use
+                `start_row = end_row = row` and `start_col = end_col = col`.
+                To delete the text in a range, use `replacement = {}`.
 
-                Prefer nvim_buf_set_lines when adding or deleting entire lines
-                only.
+                Prefer |nvim_buf_set_lines()| if you are only adding or
+                deleting entire lines.
 
                 Parameters: ~
                     {buffer}       Buffer handle, or 0 for current buffer
                     {start_row}    First line index
-                    {start_col}    First column
-                    {end_row}      Last line index
-                    {end_col}      Last column
+                    {start_col}    Starting byte offset of first line
+                    {end_row}      Last line index, inclusive
+                    {end_col}      Ending byte offset of last line, exclusive
```

Comment:
```suggestion
                    {end_col}      Ending byte offset on last line, exclusive
```

Reply:
I've used @bfredl 's suggestion, which is a superset of this.